File: Backend/sigr/src/main/java/com/alomrane/sigr/model/Utilisateur.java
package com.alomrane.sigr.model;

import com.alomrane.sigr.model.enums.RoleUtilisateur;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "utilisateurs")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(exclude = "employe")
public class Utilisateur {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String loginLdap;

    private boolean actif;

    private LocalDateTime derniereConnexion;

    @Enumerated(EnumType.STRING)
    private RoleUtilisateur role;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employe_id", unique = true)
    private Employe employe;
}

File: Frontend/types/utilisateur.ts
export type RoleUtilisateur = "ADMIN_SI" | "CHEF_SERVICE" | "EMPLOYE" | "RESPONSABLE_LOGISTIQUE";

export interface UtilisateurDto {
  id: number;
  loginLdap: string;
  actif: boolean;
  role: RoleUtilisateur;
  employeId: number;
  employeNom: string;
  department: string;
  derniereConnexion: string;
  createdAt: string;
  phone?: string;
}

export interface CreateUtilisateurRequest {
  loginLdap: string;
  actif: boolean;
  role: RoleUtilisateur;
  employeId: number;
}